sys/net/gnrc/netif/ieee802154/gnrc_netif_ieee802154.c

```diff
@@ -186,9 +177,34 @@ static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif)
             memcpy(netif->last_pkt.src, gnrc_netif_hdr_get_src_addr(hdr),
                    hdr->src_l2addr_len);
             netif->last_pkt.src_len = hdr->src_l2addr_len;
-            netif->last_pkt.seq = ieee802154_get_seq(ieee802154_hdr->data);
+            netif->last_pkt.seq = ieee802154_get_seq(mhr);
 #endif /* MODULE_GNRC_NETIF_DEDUP */
-
+#if IS_USED(MODULE_IEEE802154_SECURITY)
+            gnrc_pktsnip_t *ieee802154_aux_hdr = NULL;
+            uint8_t *auxiliary_hdr = NULL;
+            uint8_t auxiliary_hdr_size = 0;
+            uint8_t *payload = NULL;
+            uint8_t payload_size = 0;
+            uint8_t *mic = NULL;
+            uint8_t mic_size = 0;
+            if (mhr[0] & NETDEV_IEEE802154_SECURITY_EN) {
+                if (ieee802154_sec_decrypt_frame(&((netdev_ieee802154_t *)dev)->sec_ctx,
```

Comment:
since there's an ongoing migration to the Radio HAL (#14792 ), this should be ideally independent of `netdev`.

Reply:
I don´t know where else to call `ieee802154_sec_decrypt_frame / ieee802154_sec_encrypt_frame` right now.
I would also don´t mind if the security was declared as `experimental`, until the radio HAL is done.

Reply:
aaaah wait! This is `gnrc_netif_ieee802154`. Since this is still `netdev` dependent, it shouldn't be a problem here.